### ISSUE TITLE
Amending a tagged, unpushed commit causes the tag to disappear from the GUI

### ISSUE THREAD

**Issue by gbakeman** [desktop/desktop#17737]
### The problem

If you have an unpushed commit with a tag and decide to amend that commit, the tag will disappear from the commit after you complete the amendment. However, the tag is still present in the local git repository and will need to be removed by the command line interface.

### Release version

3.3.5

### Operating system

Windows 11

### Steps to reproduce the behavior

1. Make a modification - a new file, change an existing file, etc.
2. Commit the change(s)
3. Tag the commit in the History tab
4. Amend the previous commit
5. Commit your changes
6. Note that the tag has disappeared from the GUI. Attempting to re-add the tag will result in an error stating that the tag already exists.

### Log files

[2023-11-16.desktop.production.log](https://github.com/desktop/desktop/files/13380613/2023-11-16.desktop.production.log)

**Comment by steveward** [comment]
Closing since this is the expected behavior, see my comment in https://github.com/desktop/desktop/issues/18462#issuecomment-2059846352.

**Comment by gbakeman** [comment]
@steveward I didn't realize someone had opened a duplicate issue to this one, but it looks like you've received a reply explaining how this is still a problem.

I see that you linked to another feature request for searching tags, although I'm not sure if this is an ideal solution. Perhaps my workflow is different, but what I've had to do in the past is remove the tag (in CLI), then retag the amended commit. I understand how the amended commit has a different hash, but is there any reason why the UI can't automate the process of deleting the tag and reapplying it?

**Comment by steveward** [comment]
I agree we could handle this better since it is a source of confusion -- I'll reopen this and add an enhancement label for future consideration.
